core/src/main/java/org/infinispan/factories/ComponentRegistry.java

```diff
@@ -554,4 +558,12 @@ public ComponentRef<BackupReceiver> getBackupReceiver() {
       return backupReceiver;
    }
 
+   public static <K, V> ComponentRegistry of(Cache<K, V> cache) {
+      return ((InternalCache<K, V>) cache).getComponentRegistry();
+   }
+
+   public static <T> T componentOf(Cache<?, ?> cache, Class<T> type) {
```

Comment:
It would be nice to also have an overload for named components, e.g.:

```java
public static <T> T componentOf(Cache<?, ?> cache, Class<T> type, String name)
```

Reply:
Done. However I could only find a single use for that.